test/redis.test.js

```diff
@@ -0,0 +1,33 @@
+'use strict'
+
+if (process.env.CI) {
+  const { mock, before, afterEach, teardown } = require('tap')
+  const redisStoreFactory = require('connect-redis')
+  const Redis = require('ioredis')
+  const fastifySession = require('../lib/fastifySession')
+
+  const RedisStoreFactory = redisStoreFactory(fastifySession)
+
+  const client = new Redis()
+
+  class RedisStore extends RedisStoreFactory {
+    constructor () {
+      super({ client })
+    }
+  }
+
+  before(async () => {
+    await client.flushall()
+  })
+
+  afterEach(async () => {
+    await client.flushall()
+  })
+
+  teardown(() => client.disconnect())
+
+  mock('./base.test', { '../lib/store': RedisStore })
+  mock('./cookie.test', { '../lib/store': RedisStore })
+  mock('./session.test', { '../lib/store': RedisStore })
+  mock('./store.test', { '../lib/store': RedisStore })
```

Comment:
As I understand this code, you'd run the same tests but mock the full `lib/store` module. I would rather prefer an approach that did not rely on mocking - I fear this is going to be very brittle to maintain over time.

Reply:
Well it I use here mock to make it behave like proxyquire. Instead of loading the MemoryStore we actually load the real RedisStore. Does this make sense to you?

Reply:
Indeed, I don't think this would be easy to maintain long-term.

Reply:
I close this for now, as I need the changes in the actual codebase to also add file store tests. 

But I would like to know, why this simple change is not easy to maintain long-term.

Reply:
It's unscalable to the addition of more databases as running multiple dbs in the same action is a recipe for flaky tests.
It's incredibly hard to debug because it runs all the tests at the same time too.

A better approach is to set a config/db as an environment variable and kick off the full test suite for every database.